mlcomp/parallelm/mlpiper/wizard_shell.py

```diff
@@ -0,0 +1,383 @@
+
+from pypsi.shell import Shell
+from pypsi.core import Command, PypsiArgParser, CommandShortCircuit
+from pypsi.commands.help import HelpCommand
+from pypsi.commands.exit import ExitCommand
+from pypsi.ansi import AnsiCodes
+
+from parallelm.pipeline.component_info import ComponentInfo, ComponentInfoEncoder
+from parallelm.pipeline.component_connection_info import ComponentConnectionInfo
+from parallelm.pipeline.component_argument_info import ComponentArgumentInfo
+from parallelm.mlpiper.wizard_modules import MainSectionWizard, ConnectionWizard, ArgumentWizard, SaveLoadWizard
+
+import six
+import json
+import io
+
+EXIT_CODE = 127
+
+
+class CommandName:
+    COMPONENT = "comp"
+    ARGUMENT = "argument"
+    INPUT = "input"
+    OUTPUT = "output"
+    SHOW = "show"
+    SAVE = "save"
+    LOAD = "load"
+    HELP = "help"
+
+
+class CompCommand(Command):
+    def __init__(self, name, topic, brief, **kwargs):
+        self.parser = PypsiArgParser(prog=name, description=brief)
+
+        super(CompCommand, self).__init__(name=name, usage=self.parser.format_help(), topic=topic, brief=brief, **kwargs)
+
+    def run(self, shell, args):
+        try:
+            ns = self.parser.parse_args(args)
+        except CommandShortCircuit as e:
+            return e.code
+
+        component_info = shell._component_info
+        # read print_header from args
+        print_header = True
+        w = MainSectionWizard(component_info)
```

Comment:
I've noticed this kind of pattern throughout the pull requests. Usually, if you don't use the variable `w`, besides calling a single function, you don't really need to allocate it. 
You may do something like `ClassName(...).run()`

Reply:
fixed

Reply:
done